notifications/pipeline_steps/base.py

```diff
@@ -0,0 +1,25 @@
+"""Base class for all the pipeline steps."""
+
+from datetime import datetime
+import logging
+
+import pytz
+
+
+class PipelineStep(object):
+    """Implement base functionality which all the pipeline steps inherit."""
+
+    @staticmethod
+    def process_message(message):
+        """
+        Drop the message if it has expired.
+
+        Args:
+            message: NotificationsMessage object received from the previous step in the pipeline.
+        """
+        logging.info("%s has received %s %s", message.current_step, message.name, message.uuid)
+        if pytz.utc.localize(datetime.utcnow()) > message.expiration_time:
+            logging.warning("%s %s has expired. It will be dropped.", message.name, message.uuid)
+            # TODO: Implement dropping
```

Comment:
You can remove this `TODO`.  Dropping doesn't really need any further implementation - relative to the other functionality.